### ISSUE TITLE
Completion error when using Fragment in Domain Model Example

### ISSUE THREAD

**Issue by matheuslenke** [langium/langium#603]
Similar to #120 I had a completion error when I invoke the completion at the position marked with `|` below:

```
entity Comment extends |   {
    content: String
}
``` 

the request fails with the following output:

```
[Error - 4:54:42 PM] Request textDocument/completion failed.
  Message: Request textDocument/completion failed with message: EntityInfo:superType is not a valid reference id.
  Code: -32603 
```

The only difference is that my Entity rule has a fragment now, like this:

```
Entity:
    'entity' name=ID 
    EntityInfo;

fragment EntityInfo:
    ('extends' superType=[Entity:QualifiedName])? '{'
        (features+=Feature)*
    '}'
;
```

I found this in another project and tried to reproduce it in an example to make sure it wasn't anything specific from my project. When i tried to just put everything in one rule without fragments, it worked properly.

Langium version: 0.4.0
Package name: langium

## Steps To Reproduce

1. Change the Entity rule in the Domain Model example to be like above
2. Create a `.dmodel` file with the example content, and try to press ctrl+space at the correct location

## The current behavior
The completion request fails

## The expected behavior
Completion should propose the def keyword as well as the names of the already available definitions (through Evaluation > Expression).

**Comment by msujew** [comment]
We already noticed a few issues when working with fragments (and especially references within them). I tried solving this in https://github.com/langium/langium/pull/533, but the current approach in there is missing some stuff and will get refactored soonish.

**Comment by msujew** [comment]
This has been resolved with https://github.com/langium/langium/pull/623 and a fix has been released with Langium 0.5.
